lib/srv/db/ca.go

```diff
@@ -19,28 +19,49 @@ package db
 import (
 	"bytes"
 	"context"
+	"crypto/sha256"
 	"fmt"
 	"io"
 	"net/http"
 	"os"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/gravitational/trace"
-	sqladmin "google.golang.org/api/sqladmin/v1beta4"
 
 	"github.com/gravitational/teleport"
 	"github.com/gravitational/teleport/api/types"
 	awsutils "github.com/gravitational/teleport/api/utils/aws"
+	"github.com/gravitational/teleport/lib/cloud/gcp"
 	"github.com/gravitational/teleport/lib/tlsca"
 	"github.com/gravitational/teleport/lib/utils"
 )
 
+// caRenewer renewer which is going to renew cloud-based database CA.
+func (s *Server) caRenewer(ctx context.Context) {
+	schedule := s.cfg.Clock.NewTicker(caRenewInterval)
+	defer schedule.Stop()
+
+	for {
+		select {
+		case <-schedule.Chan():
+			for _, database := range s.getProxiedDatabases() {
+				if err := s.initCACert(ctx, database); err != nil {
+					s.log.WithError(err).Errorf("Failed to renew database %q CA.", database.GetName())
+				}
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
 // initCACert initializes the provided server's CA certificate in case of a
 // cloud hosted database instance.
 func (s *Server) initCACert(ctx context.Context, database types.Database) error {
 	// CA certificate may be set explicitly via configuration.
-	if len(database.GetCA()) != 0 {
+	if database.GetCA() != database.GetStatusCA() {
```

Comment:
I see we are adding this check so that if the status CA is set, we want to continue instead of returning nil here. It's not very obvious at first glance. Could we improve the readability a little bit?

Reply:
I've added a comment section above explaining this comparison. Do you think it is sufficient to make it clear?